src/main/java/daomephsta/silverfish/mixin/origin_tracing/SimpleRegistryMixin.java

```diff
@@ -13,31 +16,28 @@
 import daomephsta.silverfish.Silverfish;
 import daomephsta.silverfish.SilverfishConfig;
 import daomephsta.silverfish.origin_tracing.OriginAware;
-import net.minecraft.util.registry.MutableRegistry;
-import net.minecraft.util.registry.Registry;
-import net.minecraft.util.registry.RegistryEntry;
-import net.minecraft.util.registry.RegistryEntry.Reference;
-import net.minecraft.util.registry.SimpleRegistry;
+import net.minecraft.registry.MutableRegistry;
+import net.minecraft.registry.Registry;
+import net.minecraft.registry.entry.RegistryEntry;
+import net.minecraft.registry.entry.RegistryEntry.Reference;
+import net.minecraft.registry.SimpleRegistry;
 
 @Mixin(SimpleRegistry.class)
-public abstract class SimpleRegistryMixin<T> extends MutableRegistry<T>
+public abstract class SimpleRegistryMixin<T> implements MutableRegistry<T>
 {
-    // Required to compile, ignored by Mixin
-    private SimpleRegistryMixin()
-    {
-        super(null, null);
-    }
+
+    @Shadow @Nullable private Map<T, Reference<T>> intrusiveValueToEntry;
 
     @Inject(method = "freeze",
         at = @At(value = "NEW", target = "java.lang.IllegalStateException", ordinal = 1),
         locals = LocalCapture.CAPTURE_FAILHARD)
-    private void silverfish_improveIntrusiveHoldersError(CallbackInfoReturnable<Registry<T>> info,
-        List<RegistryEntry.Reference<?>> notAdded)
+    private void silverfish_improveIntrusiveHoldersError(CallbackInfoReturnable<Registry<T>> info)
     {
+        var notAdded = this.intrusiveValueToEntry.values().stream().toList();
```

Comment:
This is unnecessary work. The code only needs a Collection, not a List specifically.
`this.intrusiveValueToEntry.values()` is sufficient.
Also I do not use var in my codebases unless the return type is part of the right hand side.
```suggestion
      Collection<Reference<T>> notAdded = this.intrusiveValueToEntry.values();
```